packages/odyssey-react-mui/src/labs/DataView/DataView.test.tsx

```diff
@@ -0,0 +1,89 @@
+/*!
+ * Copyright (c) 2024-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import { fireEvent, render, screen, waitFor } from "@testing-library/react";
+import { DataView } from "./index";
+import {
+  data,
+  columns,
+} from "@okta/odyssey-storybook/src/components/odyssey-labs/DataView/personData";
+import { filterData } from "@okta/odyssey-storybook/src/components/odyssey-labs/DataView/dataFunctions";
+
+const getData = ({ ...props }) => {
+  return filterData({ data, ...props });
+};
+
+const testView = (
+  <DataView
+    availableLayouts={["table"]}
+    getData={getData}
+    tableLayoutOptions={{
+      columns: columns,
+    }}
+    hasSearch
+    hasPagination
+    paginationType="loadMore"
+    resultsPerPage={20}
+  />
+);
+
+describe("DataView", () => {
+  it("displays the expected number of rows by default", async () => {
+    render(testView);
+
+    await waitFor(() => {
+      const rows = screen.getAllByRole("row");
+      // one more than resultsPerPage, because of the thead row
+      expect(rows.length).toBe(21);
+    });
+  });
```

Comment:
I think these tests should resemble more how we expect our users to test, and they might have multiple tables on the page.

In that case, we'd do something like selecting the table first, then checking the rows "within" that table, but then I thought about it more, and that should probably be its own test:

```ts
render(<DataView ... />)

const tableElement = await screen.findByRole("table", { name: "" })

expect(tableElement).toBeVisible()
```

We could even try saying "I want the rows in the table similarly:

```ts
render(<DataView ... />)

const tableElement = await screen.findByRole("table", { name: "" })

const rowElements = within(tableElement).getAllByRole("row")

expect(rowElements.length).toBe(21)
```

We also need to be testing if these rows are even visible. So if we expect 21 rows, are they hidden or are they visible in the DOM?

One way is to change your selector:
```ts
const { container: tableElement } = render(<DataTable ... />)

const rowElements = within(tableElement).getAllByRole("row", { hidden: false })

expect(rowElements.length).toBe(21)
```